gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -57,17 +59,36 @@ public final Set<String> getCompletions(Set<String> repoUrls, DependencyGroup in
                 .flatMap(Optional::stream)
                 .collect(Collectors.toSet());
 
+        Set<String> resultsWithStarsIncluded = results.stream()
+                .map(result -> {
+                    int colonIndex = result.indexOf(':');
+                    if (colonIndex != -1) {
+                        return Arrays.asList(result, result.substring(0, colonIndex) + ":*");
+                    }
+                    return Collections.singletonList(result);
+                })
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
         if (parsedInput.isEmpty()) {
             return results;
         }
 
-        stopWatch.stop();
-        log.debug("Completion matching time: {} ms", stopWatch.elapsed().toMillis());
-
-        return results.stream()
+        Set<String> filteredResults = resultsWithStarsIncluded.stream()
                 .filter(result -> result.startsWith(parsedInput))
                 .map(result -> result.substring(parsedInput.length() + 1))
                 .collect(Collectors.toSet());
+
+        if (isPackageName) {
+            filteredResults = filteredResults.stream()
+                    .filter(result -> !result.contains(":"))
+                    .collect(Collectors.toSet());
+        }
+
+        stopWatch.stop();
+        log.debug("Completion matching time: {} ms", stopWatch.elapsed().toMillis());
```

Comment:
I often prefer to return where possible over variable mutation as it's a bit easier to see where things end and what will be there at the end. This is harder now that you have the stopwatch and also the two different exit paths.

You can keep it in the class, but a method that will probably be useful is some sort of `timeOperation` method e.g.

```java
<T> T timeOperation(String operation, Supplier<T> supplier) {
  Stopwatch stopWatch = Stopwatch.createStarted();
  T result = supplier.get();
  stopWatch.stop();
  log.debug("Operation '{}' took {} ms", ...);
  return result; 
}
```

Then inside, you can return whenever, however you want. It matters a tiny bit less if you do the above suggestion since you'll only have one exit point.

If things were particularly performance sensitive, you would then do something like checking whether the `logger` had `debug` mode enabled, and if not, then avoid creating all the stopwatch/logging methods. This works nicely because you wouldn't have to change your calling code. It's likely not necessary for now, but something to be aware of.

Reply:
good to know thanks!